core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -498,16 +498,31 @@ class SnappyParser(session: SnappySession)
         })
   }
 
+
+  protected final def tableValuedFunctionExpressions: Rule1[
+      Seq[Expression]] = rule {
+    '(' ~ ws ~ (expression + commaSep).? ~ ws ~ ')' ~ ws ~>
+        ((e: Any) => e.asInstanceOf[Option[Vector[Expression]]] match {
+          case Some(ve) => ve
+          case _ => Seq.empty
+        })
+  }
+
   protected final def relationFactor: Rule1[LogicalPlan] = rule {
     tableIdentifier ~ streamWindowOptions.? ~
-        (AS ~ identifier | strictIdentifier).? ~>
+    (AS ~ identifier | strictIdentifier).? ~ tableValuedFunctionExpressions.? ~>
         ((tableIdent: TableIdentifier,
-            window: Any, alias: Any) => window.asInstanceOf[Option[
+            window: Any, alias: Any, tvfe: Any) => window.asInstanceOf[Option[
             (Duration, Option[Duration])]] match {
           case None =>
-            val optAlias = alias.asInstanceOf[Option[String]]
-            updatePerTableQueryHint(tableIdent, optAlias)
-            UnresolvedRelation(tableIdent, optAlias)
+            tvfe.asInstanceOf[Option[Seq[Expression]]] match {
+              case None =>
+                val optAlias = alias.asInstanceOf[Option[String]]
+                updatePerTableQueryHint(tableIdent, optAlias)
+                UnresolvedRelation(tableIdent, optAlias)
+              case Some(exprs) =>
+                UnresolvedTableValuedFunction(tableIdent.identifier.toLowerCase(), exprs)
```

Comment:
This was required because it does a lookup in a builtInFunctions somewhere which failed if the range was coming as RANGE. Ok will use the above suggested.